File: english-hub-frontend/src/services/lessonService.ts
import axiosInstance from './axiosInstance';
import {
  LessonCreateRequest,
  LessonResponse,
  LessonUpdateRequest,
} from '@/types/lessonType';

export async function getAllLessons(
  courseId: string
): Promise<LessonResponse[]> {
  const response = await axiosInstance.get(`/courses/${courseId}/lessons`);
  return response.data.result;
}

export async function createLesson(
  courseId: string,
  lesson: LessonCreateRequest
): Promise<LessonResponse> {
  const response = await axiosInstance.post(
    `/courses/${courseId}/lessons`,
    lesson
  );
  return response.data.result;
}

export async function updateLesson(
  courseId: string,
  lessonId: string,
  lesson: LessonUpdateRequest
): Promise<LessonResponse> {
  const response = await axiosInstance.put(
    `/courses/${courseId}/lessons/${lessonId}`,
    lesson
  );
  return response.data.result;
}

export async function deleteLesson(
  courseId: string,
  lessonId: string
): Promise<string> {
  const response = await axiosInstance.delete(
    `/courses/${courseId}/lessons/${lessonId}`
  );
  return response.data.message;
}

File: english-hub-backend/src/main/java/com/example/englishhubbackend/service/LessonService.java
package com.example.englishhubbackend.service;

import com.example.englishhubbackend.dto.request.LessonCreateRequest;
import com.example.englishhubbackend.dto.request.LessonUpdateRequest;
import com.example.englishhubbackend.dto.response.LessonResponse;
import java.util.List;
import java.util.UUID;

public interface LessonService {
  List<LessonResponse> getAllLessonsFromCourse(UUID courseId);

  LessonResponse createLesson(UUID courseId, LessonCreateRequest lessonCreateRequest);

  LessonResponse updateLesson(UUID lessonId, LessonUpdateRequest lessonUpdateRequest);

  void deleteLesson(UUID lessonId);

  LessonResponse getLessonById(UUID courseId, UUID lessonId);
}
